pkg/pillar/vault/handler_zfs.go

```diff
@@ -418,7 +457,7 @@ func CreateZvolEtcd(log *base.LogObject, datasetName string) error {
 		etcdSizeBytes = 1024 * 1024 * 1024 * 10
 	}
 
-	err = zfs.CreateVolumeDataset(log, datasetName, etcdSizeBytes, "off")
+	err = zfs.CreateVaultVolumeDataset(log, datasetName, zfsKeyFile, encrypted, etcdSizeBytes, "off", uint64(4*1024))
```

Comment:
Or hardocde 4kB??

Reply:
The vault is created with all remaining space but we still want to end the volume on aligned block size so we backtrack 1MB and then align up to the block size.

Reply:
Makes sense to define a constant (e.g., etcdVolBlockSizeBytes) for this.

Reply:
Done in latest commit EtcdVolBlockSizeBytes